src/resolver/resolve_path.zig

```diff
@@ -496,68 +502,43 @@ pub fn relativeAlloc(allocator: std.mem.Allocator, from: []const u8, to: []const
     return try allocator.dupe(u8, result);
 }
 
-// This function is based on Go's volumeNameLen function
-// https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/path/filepath/path_windows.go;l=57
 // volumeNameLen returns length of the leading volume name on Windows.
+//
+// This function is based on Go's volumeNameLen function but zigified:
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/path/filepath/path_windows.go;l=57
 fn windowsVolumeNameLen(path: []const u8) struct { usize, usize } {
-    if (path.len < 2) return .{ 0, 0 };
-    // with drive letter
-    const c = path[0];
-    if (path[1] == ':') {
-        if ('a' <= c and c <= 'z' or 'A' <= c and c <= 'Z') {
-            return .{ 2, 0 };
-        }
-    }
-    // UNC
-    if (path.len >= 5 and
-        Platform.windows.isSeparator(path[0]) and
-        Platform.windows.isSeparator(path[1]) and
-        !Platform.windows.isSeparator(path[2]) and
-        path[2] != '.')
-    {
-        if (strings.indexOfAny(path[3..], "/\\")) |idx| {
-            // TODO: handle input "//abc//def" should be picked up as a unc path
-            if (path.len > idx + 4 and !Platform.windows.isSeparator(path[idx + 4])) {
-                if (strings.indexOfAny(path[idx + 4 ..], "/\\")) |idx2| {
-                    return .{ idx + idx2 + 4, idx + 3 };
-                } else {
-                    return .{ path.len, idx + 3 };
-                }
-            }
-        }
+    const parsed = std.fs.path.windowsParsePath(path);
+    if (parsed.kind == WindowsPath.Kind.None) {
+        return .{ 0, 0 };
     }
-    return .{ 0, 0 };
+    return .{ parsed.disk_designator.len, 0 };
 }
 
 pub fn windowsVolumeName(path: []const u8) []const u8 {
     return path[0..@call(.always_inline, windowsVolumeNameLen, .{path})[0]];
 }
 
+pub fn getRootPosix(path: []const u8) []const u8 {
+    return path[0..1];
+}
+
 // path.relative lets you do relative across different share drives
-pub fn windowsFilesystemRoot(path: []const u8) []const u8 {
-    if (path.len < 3)
-        return if (isSepAny(path[0])) path[0..1] else path[0..0];
-    // with drive letter
-    const c = path[0];
-    if (path[1] == ':' and isSepAny(path[2])) {
-        if ('a' <= c and c <= 'z' or 'A' <= c and c <= 'Z') {
-            return path[0..3];
-        }
+pub fn getRootWindows(path: []const u8) []const u8 {
+    const parsed = std.fs.path.windowsParsePath(path);
+    const disk_designator = parsed.disk_designator;
+    switch (parsed.kind) {
+        WindowsPath.Kind.None => return path[0..0],
+        WindowsPath.Kind.Drive => return if (path.len > 3) path[0 .. disk_designator.len + 1] else path[0..0],
+        WindowsPath.Kind.NetworkShare => return if (path.len > 4) path[0 .. disk_designator.len + 1] else path[0..0],
```

Comment:
☝️ Specifically the `NetworkShare` case needs looking at. I think it may need to be closer to the `strings.indexOfAny(path[3..], "/\\")` thing before. Second look needed :pray: for sure.